pkg/operator/controller/gateway-service-dns/controller.go

```diff
@@ -241,8 +242,28 @@ func (r *reconciler) ensureDNSRecordsForGateway(ctx context.Context, gateway *ga
 		Name:       service.Name,
 		UID:        service.UID,
 	}
+	ics := operatorv1.IngressControllerList{}
+	if err := r.cache.List(ctx, &ics); err != nil {
+		log.Error(err, "failed to list Ingress Controllers")
+		return []error{err}
+	}
 	var errs []error
 	for _, domain := range domains {
+		// Check if the domain matches one from an existing ingress controller. If so, log an error and ignore the invalid domain.
+		domainClashes := false
+		for _, ic := range ics.Items {
+			if len(ic.Status.Domain) == 0 {
+				continue
+			}
+			if domainsEqual(domain, "*."+ic.Status.Domain) {
```

Comment:
It's possible for a gateway hostname to be non-wildcard.  Is there a case where you had to prepend `*.`  to the ingress controller domain to make this work?

Reply:
It would be a little faster to just do `strings.Compare(domain, ic.Status.Domain+".") == 0`

Reply:
While the gateway hostname could be either a wildcard or not, `ic.Status.Domain` is always non-wildcard, but the DNS record created for it is always a wildcard (`desiredWildcardDNSRecord` always adds `*.` before and `.` after the domain, shown here: https://github.com/openshift/cluster-ingress-operator/blob/master/pkg/resources/dnsrecord/dns.go#L108). We need to compare the domain names on the DNS records, so `ic.Status.Domain` does need `*.` prepended.